File: frontend/metavalley/features/startups/startup.schema.ts
import { z } from "zod";

export const stageOptions = [
  { label: "Ideia", value: "idea" },
  { label: "MVP", value: "mvp" },
  { label: "Lançada", value: "launched" },
] as const;

export const revenueModelOptions = [
  { label: "Assinatura", value: "subscription" },
  { label: "SaaS", value: "saas" },
  { label: "Marketplace", value: "marketplace" },
  { label: "Freemium", value: "freemium" },
  { label: "Pagamento por uso", value: "pay_per_use" },
  { label: "Licenciamento", value: "licensing" },
  { label: "Publicidade", value: "advertising" },
  { label: "E-commerce", value: "e_commerce" },
  { label: "Serviço", value: "service" },
  { label: "Hardware", value: "hardware" },
  { label: "Outro", value: "other" },
] as const;

export const createStartupSchema = z.object({
  name: z.string().min(1, "O nome da startup é obrigatório.").max(255),
  description: z.string().optional(),
  problem: z.string().optional(),
  solution: z.string().optional(),
  segment: z.string().max(255).optional(),
  target_location: z.string().max(255).optional(),
  stage: z.enum(["idea", "mvp", "launched"]),
  primary_revenue_model: z
    .enum([
      "subscription",
      "saas",
      "marketplace",
      "freemium",
      "pay_per_use",
      "licensing",
      "advertising",
      "e_commerce",
      "service",
      "hardware",
      "other",
    ])
    .optional(),
  revenue_model_details: z.string().optional(),
});

export type CreateStartupFormData = z.infer<typeof createStartupSchema>;

export const startupSteps: { title: string; fields: (keyof CreateStartupFormData)[] }[] = [
  { title: "Informações básicas", fields: ["name", "description", "target_location"] },
  { title: "Problema e solução", fields: ["problem", "solution"] },
  {
    title: "Modelo de negócio",
    fields: ["segment", "stage", "primary_revenue_model", "revenue_model_details"],
  },
];

File: backend/app/schemas/startup_schema.py
from datetime import datetime
from enum import Enum
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class StartupStage(str, Enum):
    IDEA = "idea"
    MVP = "mvp"
    LAUNCHED = "launched"


class RevenueModel(str, Enum):
    SUBSCRIPTION = "subscription"
    SAAS = "saas"
    MARKETPLACE = "marketplace"
    FREEMIUM = "freemium"
    PAY_PER_USE = "pay_per_use"
    LICENSING = "licensing"
    ADVERTISING = "advertising"
    E_COMMERCE = "e_commerce"
    SERVICE = "service"
    HARDWARE = "hardware"
    OTHER = "other"


class StartupStatus(str, Enum):
    ACTIVE = "active"
    ARCHIVED = "archived"


class StartupBase(BaseModel):
    name: str = Field(..., max_length=255, description="Nome da startup")
    description: Optional[str] = None
    problem: Optional[str] = None
    solution: Optional[str] = None
    segment: Optional[str] = Field(None, max_length=255)
    target_location: Optional[str] = Field(None, max_length=255)
    stage: StartupStage = StartupStage.IDEA
    primary_revenue_model: Optional[RevenueModel] = None
    revenue_model_details: Optional[str] = None
    status: StartupStatus = StartupStatus.ACTIVE


class StartupCreate(StartupBase):
    pass


class StartupUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=255)
    description: Optional[str] = None
    problem: Optional[str] = None
    solution: Optional[str] = None
    segment: Optional[str] = Field(None, max_length=255)
    target_location: Optional[str] = Field(None, max_length=255)
    stage: Optional[StartupStage] = None
    primary_revenue_model: Optional[RevenueModel] = None
    revenue_model_details: Optional[str] = None
    status: Optional[StartupStatus] = None


class StartupResponse(StartupBase):
    id: UUID
    user_id: UUID
    image_url: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)